tests/python/gaia-ui-tests/gaiatest/tests/functional/email/test_IMAP_email_notification.py

```diff
@@ -64,6 +64,7 @@ def test_IMAP_email_notification(self):
         self.apps.switch_to_displayed_app()
 
         # check if the sender's email address is fine
+        self.wait_for_condition(lambda m: email.senders_email == mock_email.senders_email)
```

Comment:
I agree with Johan here, these are redundant. I'd either remove the assertion and move the error message to the wait (though don't know if it can be templated if you do that), or trap the error on the wait and then assert inside the error handler with a clean message. 

Personally, I'd probably do the former. Our waits are all pretty much assertions in disguise.